packages/api/src/payments/stripePaymentProvider.ts

```diff
@@ -119,29 +121,51 @@ export class StripePaymentProvider extends BasePaymentProvider {
         )
       }
     }
+    let intent
+    let errorCode
+    try {
+      intent = await this.stripe.paymentIntents.create({
+        amount: invoice.items.reduce(
+          (prevItem, currentItem) => prevItem + currentItem.amount * currentItem.quantity,
+          0
+        ),
+        ...(customerID && paymentMethodID
+          ? {
+              confirm: true,
+              customer: customerID,
+              off_session: true,
+              // payment_method: paymentMethodID,
+              payment_method: 'pm_card_chargeDeclinedExpiredCard',
+              payment_method_types: ['card']
+            }
+          : {}),
+        currency: 'chf',
+        // description: props.invoice.description, TODO: convert to text
+        ...(saveCustomer ? {setup_future_usage: 'off_session'} : {}),
+        metadata: {
+          paymentID: paymentID,
+          mail: invoice.mail
+        }
+      })
+    } catch (error) {
```

Comment:
Maybe from line 127 to 148 can be broken into different variable for improved maintainability